src/modules/land_detector/LandDetector.cpp

```diff
@@ -101,14 +101,40 @@ void LandDetector::_cycle()
 	_check_params(false);
 	_orb_update(ORB_ID(actuator_armed), _armingSub, &_arming);
 	_update_topics();
-	_update_state();
 
-	const bool landDetected = (_state == LandDetectionState::LANDED);
-	const bool freefallDetected = (_state == LandDetectionState::FREEFALL);
-	const bool maybe_landedDetected = (_state == LandDetectionState::MAYBE_LANDED);
-	const bool ground_contactDetected = (_state == LandDetectionState::GROUND_CONTACT);
+
+	const bool freefall_state_init = _get_freefall_state();
+	const bool landed_state_init = _get_landed_state();
+	const bool maybe_landed_state_init = _get_maybe_landed_state();
+	const bool ground_contact_state_init = _get_ground_contact_state();
+
+	_freefall_hysteresis.update(_get_freefall_state());
+	_landed_hysteresis.update(_get_landed_state());
+	_maybe_landed_hysteresis.update(_get_maybe_landed_state());
+	_ground_contact_hysteresis.update(_get_ground_contact_state());
+
+	const bool landDetected = _landed_hysteresis.get_state();
+	const bool freefallDetected = _freefall_hysteresis.get_state();
+	const bool maybe_landedDetected = _maybe_landed_hysteresis.get_state();
+	const bool ground_contactDetected = _ground_contact_hysteresis.get_state();
 	const float alt_max = _get_max_altitude();
 
+	if (landDetected != landed_state_init) {
+		PX4_ERR("landDetected: %d landed_state_init: %d", landDetected, landed_state_init);
+	}
```

Comment:
These if statements are probably not what you want.
You will get an PX4_ERR message every single time the get_..._state() is true but the corresponding hysteresis not.